### PR TITLE
fix: improve code readability in Application.cpp

### PR DESCRIPTION
**Pull request by Scrumplex**
This looks really complicated. But it's fairly easy. Just look at it in a split diff.

### CODE REVIEW

**Review by timoreo22 — APPROVED** [review]
Looks good !

**Review by d-513 — APPROVED** [review]
much better

**Review by txtsd — APPROVED** [review]

File: launcher/Application.cpp [hunk 1]
```diff
@@ -285,12 +285,29 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
...
     }
+
     m_instanceIdToLaunch = args["launch"].toString();
     m_serverToJoin = args["server"].toString();
     m_profileToUse = args["profile"].toString();
     m_liveCheck = args["alive"].toBool();
     m_zipToImport = args["import"].toUrl();
 
+    // error if --launch is missing with --server
+    if(m_instanceIdToLaunch.isEmpty() && !m_serverToJoin.isEmpty())
```
Comment (ghost) [inline launcher/Application.cpp]: why can't you combine these two

**Review by ghost — CHANGES_REQUESTED** [review]
nitpick

File: launcher/Application.cpp (hunk 1 above)
Comment (Scrumplex) [reply launcher/Application.cpp]: We could. I just wanted to keep the original error messages.

Comment (ghost) [reply launcher/Application.cpp]: I see no problem with combining them in that case. We're finally completely diverting from MultiMC and there's currently no need to keep its standards, messages, etc.

except for cat

File: launcher/Application.cpp [hunk 2]
```diff
@@ -285,12 +285,21 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
...
     m_instanceIdToLaunch = args["launch"].toString();
     m_serverToJoin = args["server"].toString();
     m_profileToUse = args["profile"].toString();
     m_liveCheck = args["alive"].toBool();
     m_zipToImport = args["import"].toUrl();
 
+    // error if --launch is missing with --server or --profile
+    if((!m_serverToJoin.isEmpty() || !m_profileToUse.isEmpty()) && m_instanceIdToLaunch.isEmpty())
+    {
+        std::cerr << "--server and --profile can only be used in combination with --launch!" << std::endl;
```
Comment (ghost) [inline launcher/Application.cpp:298]: did you mean --account

**Review by ghost — CHANGES_REQUESTED** [review]

File: launcher/Application.cpp (hunk 2 above)
Comment (Scrumplex) [reply launcher/Application.cpp:298]: Nope.
https://github.com/PolyMC/PolyMC/blob/6b4469c6cce517dabe967c12b30648e0c41f12d5/launcher/Application.cpp#L244-L247

**Review by ghost — APPROVED** [review]

**Review by LennyMcLennington — APPROVED** [review]

**Pull request merged**